sql/catalyst/src/test/scala/org/apache/spark/sql/connector/catalog/InMemoryBaseTable.scala

```diff
@@ -60,6 +62,27 @@ abstract class InMemoryBaseTable(
     val numRowsPerSplit: Int = Int.MaxValue)
   extends Table with SupportsRead with SupportsWrite with SupportsMetadataColumns {
 
+  protected var currentTableVersion: Int = 0
+  private var validatedTableVersion: String = null
+
+  override def currentVersion(): String = currentTableVersion.toString
+
+  def setCurrentVersion(version: String): Unit = {
+    currentTableVersion = version.toInt
+  }
+
+  def increaseCurrentVersion(): Unit = {
+    currentTableVersion += 1
+  }
+
+  def validatedVersion(): String = {
```

Comment:
why do we need it? The added test doesn't even check it.

Reply:
This is for testing the `validatedTableVersion` from https://github.com/apache/spark/blob/master/sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableChange.java#L305

It is checked in https://github.com/apache/spark/pull/50806/files#diff-90a3aef257aa16eaa7c57ff4d99ea63e20ecd4883c2c2ac9358ce80ada4e96e6R210

I will add comments